src/lib/correlate.rs

```diff
@@ -6,11 +6,14 @@ pub fn correlate<'a>(data: &BinaryTimeSeries, possibilities: &'a [BinaryTimeSeri
     let mut correlations = Vec::new();
 
     for poss in possibilities {
-        let (xs, ys) = pairinate(poss, &data);
-        correlations.push(CorrelatedTimeSeries{
-          series: poss,
-          correlation: pearson_correlation_coefficient(&xs, &ys) as f32
-        });
+        let pairing = pairinate(poss, &data);
+        match pairing{
+            Some((xs, ys)) => correlations.push(CorrelatedTimeSeries{
+                series: poss,
+                correlation: pearson_correlation_coefficient(&xs, &ys) as f32
+            }),
+            _ => println!("Skipping non-overlapping dataset")
```

Comment:
Might be good to remove this for when we run this with lots of data sets so that it doesn't spam tons of lines to logs. replace the match with `if let Some((xs, ys)) = pairinate(poss, &data)`.